core/dragging/dragger.ts

```diff
@@ -4,65 +4,132 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import {IDragTarget} from '../interfaces/i_drag_target.js';
+import {IDeletable, isDeletable} from '../interfaces/i_deletable.js';
 import {IDragger} from '../interfaces/i_dragger.js';
 import {IDraggable} from '../interfaces/i_draggable.js';
 import {Coordinate} from '../utils/coordinate.js';
 import {WorkspaceSvg} from '../workspace_svg.js';
+import {ComponentManager} from '../component_manager.js';
+import {IDeleteArea} from '../interfaces/i_delete_area.js';
 
 export class Dragger implements IDragger {
-  /** Starting location of the draggable, in workspace coordinates. */
   private startLoc: Coordinate;
 
+  private dragTarget: IDragTarget | null = null;
+
   constructor(
     private draggable: IDraggable,
     private workspace: WorkspaceSvg,
   ) {
-    this.startLoc = draggable.getLocation();
+    this.startLoc = draggable.getRelativeToSurfaceXY();
   }
 
-  /**
-   * Handles any drag startup.
-   *
-   * @param e PointerEvent that started the drag.
-   */
+  /** Handles any drag startup. */
   onDragStart(e: PointerEvent) {
     this.draggable.startDrag(e);
   }
 
   /**
-   * Handles dragging, including calculating where the element should
-   * actually be moved to.
+   * Handles calculating where the element should actually be moved to.
    *
-   * @param e PointerEvent that continued the drag.
-   * @param totalDelta The total distance, in pixels, that the mouse
-   *     has moved since the start of the drag.
+   * @param totalDelta The total amount in pixel coordinates the mouse has moved
+   *     since the start of the drag.
    */
   onDrag(e: PointerEvent, totalDelta: Coordinate) {
-    this.draggable.drag(this.newLoc(totalDelta), e);
+    this.moveDraggable(e, totalDelta);
+
+    // Must check `wouldDelete` before calling other hooks on drag targets
+    // since we have documented that we would do so.
+    if (isDeletable(this.draggable)) {
+      (this.draggable as AnyDuringMigration).setDeleteStyle(
```

Comment:
Same for all uses of `AnyDuringMigration` in this file.

Reply:
These are going to get cleaned up after the interfaces for deletable things and selectable things are fixed. Right now the interfaces are kind of... wrong. So we have to do things like this.

Before it worked because we weren't really using the interfaces, we were only interacting with concrete classes.